src/site/resources/js/redirect.js

```diff
@@ -0,0 +1,39 @@
+window.onload = () => {
+    const currentUrl = window.location.href;
+    const urlObj = new URL(currentUrl);
+    const pathSegments = urlObj.pathname.split("/");
+    const lastSegment = pathSegments[pathSegments.length - 1];
+    const extractedUrl =  `/${lastSegment}${urlObj.hash}`;
+
+    if (currentUrl === "https://checkstyle.org/checks/") {
```

Comment:
endsWith("/checks/")

Reply:
done. 

Had to create `index.xml` under `checks` directory to make it work. 

https://github.com/checkstyle/checkstyle/blob/c9a74285db34d2963947abf6effa07f610334335/src/site/resources/js/redirect.js#L3-L6

Reply:
https://github.com/checkstyle/checkstyle/blob/c8397ba9975a92001c25d9cae4232b5cc9e07685/src/site/resources/js/redirect.js#L3-L6

made the redirection relative.